MoltenVK/MoltenVK/Commands/MVKCommandPipelineStateFactoryShaderSource.h

```diff
@@ -547,5 +547,14 @@ kernel void accumulateOcclusionQueryResults(device VisibilityBuffer& dest [[buff
     if (dest.count < max(oldDestCount, src.count)) { dest.countHigh++; }                                        \n\
 }                                                                                                               \n\
                                                                                                                 \n\
+kernel void convertUint8Indices(device uint8_t* src [[ buffer(0) ]],                                            \n\
+                                device uint16_t* dst [[ buffer(1) ]],                                           \n\
+                                constant uint32_t& numIndices [[ buffer(2) ]]) {                                \n\
+    for (uint32_t i = 0; i < numIndices; i++) {                                                                 \n\
```

Comment:
This would be better structured to use GPU parallelism, with `thread_position_in_grid`. Then you could skip the single-thread loop entirely, and let the GPU do what it does best. 

See for example `cmdFillBuffer()` in this same file.